### ISSUE TITLE
Link to console when running on Openshift

### ISSUE THREAD

**Issue by cfcosta** [kiali/kiali#1957]
JIRA Issue: https://issues.jboss.org/browse/OSSM-4

Goal:
Provide easy access to the service mesh console to users with access to namespaces in the mesh by creating a link from OpenShift UI for the managed namespaces.

Problem:
The Service Mesh console (Kiali) will be deployed in one namespace, and the application components that are managed by the service mesh will be deployed into N other namespaces. Users administering an application component who want to access the service mesh console will not know in what namespace the Kiali console is running. Information on which namespace the Kiali console is in will need to be communicated out of band unless a link is provided in the managed namespace.

Why is this important:
The value of the console is zero if users cannot locate and access it.

Dependencies (internal and external):
OpenShift console extensibility. At this time we do not believe there is a proper extension point for us to implement.

Prioritized epics + deliverables (in scope / not in scope):
As a developer I want a link to the Service Mesh Console in every project that's part of the mesh so that I do not have to navigate out of my project to get to the console

**Comment by lucasponce** [comment]
Minor comment.
Please, add a link from downstream -> upstream issue in the OSSM-4.
Then Program/Product Managet can browse to this one.

**Comment by abonas** [comment]
> Dependencies (internal and external):
> OpenShift console extensibility. At this time we do not believe there is a proper extension point for us to implement.

@cfcosta  this part is probably redundant in upstream issue?

**Comment by cfcosta** [comment]
@abonas that has been updated since the issue has been created, OCP 4.3 is developing an extension point for this.

**Comment by lucasponce** [comment]
Please, let's progress on this issue. 
It's part of our next release and today it's in the radar with a potential delay.

Let's use this work as base:
https://github.com/openshift/console/pull/3176

And let's prepare a draft with the Maistra/Kiali part, according with the info that is merged in openshift there will be needed a CR for this.

From Maistra/Kiali side we can start prototyping how when a new namespace is added to the mesh that the operator should create a CR representing this link with the project.

It doesn't matter if that CR is the final one needed for OCP 4.3 or we can start working on this with a demo one but I'm sure that we will need to check / resolve issues at Maistra/Kiali side first.

**Comment by jshaughn** [comment]
@cfcosta I'm removing the Waiting on External label.  I think progress can be made.  As far as I know the support has been merged into 4.3 and is available in a nightly build.  Also, on our side we should be able to work on the needed support.  You may want to talk with @jmazzitelli (before the break) regarding any operator work.
